### PR TITLE
Fix description for Twitch auth

### LINKED ISSUE #200: TWITCH_CONFIGURATION requires extra tokenRequestParams to be successful
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the bug has not already been reported

### Fastify version

4.15.0

### Plugin version

7.0.1

### Node.js version

18.13.0

### Operating system

macOS

### Operating system version (i.e. 20.04, 11.3, 10)

13.3.1 

### Description

In order to make a successful authentication call to Twitch, the plugin requires extra configuration that is not documented.

Here's an example of a successful authentication call, passing in extra `tokenRequestParams`:

Note:
The code below assumes that environment variables for the "Twitch Client ID" and "Twitch Secret" have been stored as secrets with the GitHub repository, and configured in `platformatic.service.json`

```js
  await app.register(fastifyOauth2, {
    name: "twitchOAuth2",
    scope: ["required:scopes"],
    credentials: {
      client: {
        id: opts.twitch_client_id,
        secret: opts.twitch_secret,
      },
      auth: fastifyOauth2.TWITCH_CONFIGURATION,
    },
    tokenRequestParams: {
      client_id: opts.twitch_client_id,
      client_secret: opts.twitch_secret,
    },
    startRedirectPath: "/twitch/login",
    callbackUri: "https://domain.tld/callback"
  });
```

### Steps to Reproduce

Attempt to authenticate without passing in `tokenRequestParams` and you'll make a `400 Bad Request` with the message `missing client id`.

### Expected Behavior

I would expect either the code example above to be documented, or for the plugin to automatically pass the client ID and secret through automatically, so the user only has to declare them once.

### PR DESCRIPTION
**Pull request by mcollina**
Fixes #200 

#### Checklist

- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Comment by mcollina** [comment]
cc @whitep4nth3r

File: README.md [hunk 1]
```diff
@@ -259,6 +259,36 @@ declare module 'fastify' {
...
 ```
 
+## Provider Quirks
+
+This is a list of known quirks of some providers that needed additional work to get everything set up correctly.
+
+### Twitch
+
+Twitch requires that the request for a token in the oauth2 flow contains the `client_id` and `client_secret` property.
+Unfortunately, this library cannot set this up automatically. Howevever, it's easy to configure:
```
Comment (whitep4nth3r) [inline README.md:269]: Suggested change. (Sometimes it's good to point out what you're doing in both the text and the code example, as I know some devs read one or the other.)

"When authenticating via Twitch, include the `client_id` and `client_secret` in `tokenRequestParams`:"

**Review by whitep4nth3r — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -259,6 +259,36 @@ declare module 'fastify' {
 }
 ```
 
+## Provider Quirks
+
+This is a list of known quirks of some providers that needed additional work to get everything set up correctly.
```
Comment (Fdawgs) [inline README.md:264]:
```suggestion
The following providers require additional work to be set up correctly.   
```

File: README.md (hunk 1 above)
Comment (mcollina) [reply README.md]: thanks, applied!

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by Fdawgs — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -259,6 +259,37 @@ declare module 'fastify' {
...
 ```
 
+## Provider Quirks
+
+The following providers require additional work to be set up correctly.   
+
+### Twitch
+
+Twitch requires that the request for a token in the oauth2 flow contains the `client_id` and `client_secret` property.
+Unfortunately, this library cannot set this up automatically. When authenticating via Twitch, include the
```
Comment (Uzlopak) [inline README.md:269]: "Unfortunately"? Is there a possibility to set this up automatically? I dont think so. So I would remove that sentence with unforrtunately.

@Fdawgs You are a native speaker of traditional english. Wdyt?

Comment (Fdawgs) [reply README.md:269]: Still valid, it is unfortunate it can't be setup automatically. 😊

**Review by Fdawgs — CHANGES_REQUESTED** [review]

**Review by whitep4nth3r — APPROVED** [review]

**Review by Fdawgs — APPROVED** [review]

**Pull request merged**